dashboard/app/helpers/levels_helper.rb

```diff
@@ -261,10 +263,37 @@ def weblab_options
     # Ensure project_template_level allows start_sources to be overridden
     app_options[:level]['startSources'] = @level.try(:project_template_level).try(:start_sources) || @level.start_sources
 
+    # Process level view options
+    level_overrides = level_view_options(@level.id).dup
+    if level_options['embed'] || level_overrides[:embed]
+      level_overrides[:hide_source] = true
+      level_overrides[:show_finish] = true
+    end
+    if level_overrides[:embed]
+      view_options(no_header: true, no_footer: true, white_background: true)
+    end
+
+    view_options(has_contained_levels: @level.try(:contained_levels).present?)
+
+    # Add all level view options to the level_options hash
+    level_options.merge! level_overrides.camelize_keys
+
     app_options.merge! view_options.camelize_keys
+
+    # Move these values up to the app_options hash
+    %w(hideSource share embed).each do |key|
+      if level_options[key]
+        app_options[key.to_sym] = level_options.delete key
+      end
+    end
+
     app_options[:app] = 'weblab'
     app_options[:baseUrl] = Blockly.base_url
 
+    if (@game && @game.owns_footer_for_share?) || @is_legacy_share
+      app_options[:copyrightStrings] = build_copyright_strings
+    end
+
```

Comment:
All of this new code has been selectively copied from `blockly_options` to `weblab_options`. Further refactoring may be needed to avoid the duplication.